substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -636,19 +635,16 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
-	/// Indices of validators that have offended in the active era and whether they are currently
-	/// disabled.
+	/// Indices of validators that have offended in the active era. The offenders are disabled for a
+	/// whole era. For this reason they are kept here - only staking pallet knows about eras. No
```

Comment:
Maybe we should add here that `[T::DisablingStrategy]` implementor defines the max/thresholds of disabled validators instead?

Reply:
Also, this seems to be a good invariant to have a as `try_state` check!